doc/lspconfig.txt

```diff
@@ -1,242 +1,628 @@
+  detected but is not based on a flaw in the lspconfig path utilities.
+- The server is launching, but you believe that the default settings,
+  initialization options, or command arguments are suboptimal and should be
+  replaced based on your understanding of the server documentation.
+
+All bugs pertaining to errors in core neovim client should be reported to the
+neovim core issue tracker. All bugs pertaining to plugins should be reported
+to the respective plugin. All missing features in a language server should be
+reported to the upstream language server issue tracker.
+
+For debugging lspconfig issues the most common mistakes users make are:
+
+  - The language server is not installed or is otherwise not executable.
+    lspconfig does not install language servers for you. Ensure the `cmd`
+    defined in `server_configurations.md` is executable from the command
+    line. If the absolute path to the binary is not supplied in `cmd`, ensure
+    it is on your PATH.
+  - No root detected. Lspconfig is built around the concept of projects. See
+    |lspconfig-root-directory| for more details. Most of the time,
+    initializing a git repo will suffice.
+  - Misconfiguration. Often times users will override `cmd`, `on_init`, or
+    `handlers`. Ensure that you debug by using a stock configuration to ensure
+    your customizations are not introducing issues.
+
+|LspInfo| provides an overview of your active and configured language servers
+which can be useful for debugging.
+
+Note, that it will not report any configuration changes applied in
+`on_new_config`.
+
+==============================================================================
+LOGGING                                                      *lspconfig-logging*
+
+When debugging language servers, it is helpful to enable additional logging in
+the built-in client, specifically considering the RPC logs.
+>
+  vim.lsp.set_log_level 'trace'
+  if vim.fn.has 'nvim-0.5.1' == 1 then
+    require('vim.lsp.log').set_format_func(vim.inspect)
+  end
+<
+Attempt to run the language server, and open the log with:
+
+>
+    :lua vim.cmd('e'..vim.lsp.get_log_path())
+<
+Note, `ERROR` messages which include `stderr` only indicate that the log was
+sent to `stderr`. Many servers counter-intuitively send innocuous messages
+via stderr.
+
+==============================================================================
+SCOPE                                                         *lspconfig-scope*
+
+lspconfig is a community effort to create default configurations that fit
+within the scope of an official plugin for neovim. All features that are not
+strictly providing default configurations for language servers will be removed
+from lspconfig in time. The power of the neovim lsp ecosystem is in the
+composability and flexibility of integrating multiple plugins which maximizes
+user choice and freedom.
+
+lspconfig also opts to adhere strictly to the lsp specification, with some
```

Comment:
On a related note, maybe use "`lspconfig`" generally? Just having this in lowercase even at the start of a sentence is a bit irritating. (Very small nit; feel free to ignore.)